server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/processors/FileOperationProcess.java

```diff
@@ -175,17 +137,15 @@ private void checkFileOpenedForWrite(
         .getVariableDefinitionNode()
         .ifPresent(
             defNode ->
-                node.getNearestParentByType(NodeType.PROGRAM)
-                    .ifPresent(program ->
-                        checkFileOpenedForWrite(program, node, expectedFileKind, ctx, messageTemplate, defNode)));
+                Optional.ofNullable(ctx.getCurrentProgramNode()).ifPresent(program ->
```

Comment:
Haven't we already established the `ctx.getCurrentProgramNode() != null` at the very beginning?